src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs

```diff
@@ -223,20 +223,50 @@ protected override CompletionItemRules GetCompletionItemRules(ImmutableArray<(IS
                     item = SymbolCompletionItem.AddShouldProvideParenthesisCompletion(item);
                 }
             }
-            else if (symbol.IsKind(SymbolKind.NamedType) || symbol is IAliasSymbol aliasSymbol && aliasSymbol.Target.IsType)
+            else if (shouldProvideParenthesisCompletion(symbol, out var includeDotCommitCharacter))
             {
                 // If this is a type symbol/alias symbol, also consider appending parenthesis when later, it is committed by using special characters,
                 // and the type is used as constructor
+                // We add a flag to disallow automatic parenthesis if the type has a nested type AND the trigger character is '.', in which case the user may want
+                // to access the nested type. See https://github.com/dotnet/roslyn/issues/51629
                 if (context.IsObjectCreationTypeContext)
-                    item = SymbolCompletionItem.AddShouldProvideParenthesisCompletion(item);
+                    item = SymbolCompletionItem.AddShouldProvideParenthesisCompletion(item, includeDotCommitCharacter);
             }
 
             return item;
+
+            static bool shouldProvideParenthesisCompletion(ISymbol? symbol, out bool includeDotCommitCharacter)
+            {
+                ITypeSymbol? typeSymbol;
+                if (symbol.IsKind(SymbolKind.NamedType))
+                {
+                    typeSymbol = (ITypeSymbol)symbol;
+                }
+                else if (symbol is IAliasSymbol aliasSymbol && aliasSymbol.Target.IsType)
+                {
+                    typeSymbol = (ITypeSymbol)aliasSymbol.Target;
+                }
+                else
+                {
+                    includeDotCommitCharacter = false;
+                    return false;
+                }
+
+                includeDotCommitCharacter = typeSymbol.GetTypeMembers().IsEmpty;
```

Comment:
do we need to check if htese are accessible?  i mean, lots of types from metadata might have an inaccessible nested type.  having that change things for the user experience seems strange.